### PR TITLE
Update installer

### PR DESCRIPTION
**Pull request by rohitagarwal003**
- Removes redundant `nvidia-modprobe` command which is done in `post_installation_sequence` as well.
- Removes redundant environment variable.
- Makes installation faster by downloading only the driver.

Someone with access to gcr.io/google_containers need to build and push the new image and update the sha in the daemonset.yaml. I tested this by using my personal registry.

### CODE REVIEW

File: cos-nvidia-gpu-installer/installer.sh [hunk 1]
```diff
@@ -47,13 +47,12 @@ BIN_OUTPUT_DIR="${BASE_DIR}/bin"
 KERNEL_SRC_DIR="/lakitu-kernel"
 
 NVIDIA_DRIVER_DIR="/nvidia"
-NVIDIA_DRIVER_VERSION="375.26"
+NVIDIA_DRIVER_VERSION="375.66"
```
Comment (Amey-D) [inline cos-nvidia-gpu-installer/installer.sh]: Given the lack of versioning in the driver installer, WDYT about keeping the driver version consistent with the one shipped as part of the CUDA 8 installer?

File: cos-nvidia-gpu-installer/installer.sh [hunk 2]
```diff
@@ -47,13 +47,12 @@ BIN_OUTPUT_DIR="${BASE_DIR}/bin"
...
-NVIDIA_CUDA_MD5SUM="33e1bd980e91af4e55f3ef835c103f9b"
-NVIDIA_CUDA_PKG_NAME="cuda_8.0.61_375.26_linux.run"
-NVIDIA_DRIVER_PKG_NAME="NVIDIA-Linux-x86_64-375.26.run"
+# Source: https://www.nvidia.com/Download/index.aspx?lang=en-us
+NVIDIA_DRIVER_URL="https://us.download.nvidia.com/XFree86/Linux-x86_64/${NVIDIA_DRIVER_VERSION}/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
+NVIDIA_DRIVER_MD5SUM="9cf2c0a342a088211f25c5d8137097e6"
+NVIDIA_DRIVER_PKG_NAME="NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
 
 check_nvidia_device() {
     lspci
```
Comment (Amey-D) [inline cos-nvidia-gpu-installer/installer.sh]: Unrelated to this PR, but why do we have this `lspci` command?

File: cos-nvidia-gpu-installer/installer.sh [hunk 3]
```diff
@@ -83,19 +82,16 @@ prepare_kernel_source() {
...
+    local pkg_name="${NVIDIA_DRIVER_PKG_NAME}"
+    local url="${NVIDIA_DRIVER_URL}"
     local log_file_name="${NVIDIA_DRIVER_DIR}/nvidia-installer.log"
 
     mkdir -p "${NVIDIA_DRIVER_DIR}"
     pushd "${NVIDIA_DRIVER_DIR}"
 
-    echo "Downloading Nvidia CUDA package from ${url} ..."
+    echo "Downloading Nvidia driver from ${url} ..."
     curl -L -s "${url}" -o "${pkg_name}"
```
Comment (Amey-D) [inline cos-nvidia-gpu-installer/installer.sh]: Could you please add -S option to the curl command?

File: cos-nvidia-gpu-installer/installer.sh [hunk 4]
```diff
@@ -47,13 +47,12 @@ BIN_OUTPUT_DIR="${BASE_DIR}/bin"
...
+NVIDIA_DRIVER_VERSION="375.66"
 
-# Source: https://developer.nvidia.com/cuda-downloads
-NVIDIA_CUDA_URL="https://developer.nvidia.com/compute/cuda/8.0/Prod2/local_installers/cuda_8.0.61_375.26_linux-run"
-NVIDIA_CUDA_MD5SUM="33e1bd980e91af4e55f3ef835c103f9b"
-NVIDIA_CUDA_PKG_NAME="cuda_8.0.61_375.26_linux.run"
-NVIDIA_DRIVER_PKG_NAME="NVIDIA-Linux-x86_64-375.26.run"
+# Source: https://www.nvidia.com/Download/index.aspx?lang=en-us
+NVIDIA_DRIVER_URL="https://us.download.nvidia.com/XFree86/Linux-x86_64/${NVIDIA_DRIVER_VERSION}/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
+NVIDIA_DRIVER_MD5SUM="9cf2c0a342a088211f25c5d8137097e6"
```
Comment (Amey-D) [inline cos-nvidia-gpu-installer/installer.sh]: WDYT about moving `NVIDIA_DRIVER_VERSION` and `NVIDIA_DRIVER_MD5SUM` to the yaml file as env vars?

File: cos-nvidia-gpu-installer/installer.sh (hunk 3 above)
Comment (rohitagarwal003) [reply cos-nvidia-gpu-installer/installer.sh]: Done.

File: cos-nvidia-gpu-installer/installer.sh (hunk 1 above)
Comment (rohitagarwal003) [reply cos-nvidia-gpu-installer/installer.sh]: Makes sense. Done.

File: cos-nvidia-gpu-installer/installer.sh (hunk 2 above)
Comment (rohitagarwal003) [reply cos-nvidia-gpu-installer/installer.sh]: I don't know. Maybe @vishh knows.

File: cos-nvidia-gpu-installer/installer.sh (hunk 4 above)
Comment (rohitagarwal003) [reply cos-nvidia-gpu-installer/installer.sh]: While I like the idea of not having to bake new images for driver version changes, let's keep the daemonset as dumb as possible for now. We can do this change in a different PR.

**Comment by vishh** [comment]
@Amey-D did you get to validate the new driver patch?

**Review by vishh — APPROVED** [review]

File: cos-nvidia-gpu-installer/installer.sh (hunk 2 above)
Comment (vishh) [reply cos-nvidia-gpu-installer/installer.sh]: IIRC, it is to check if lspci is available within the sandbox that's running the script. If the script is only going to deployed within a controlled container, we dont need it.

Comment (rohitagarwal003) [reply cos-nvidia-gpu-installer/installer.sh]: Looks like it is to make sure that we exit with non-zero exit code if `lspci` doesn't exist.

If `lspci` doesn't exist, we exit with non-zero code.
If `lspci` exists but nvidia device doesn't exist, we exit with zero code.
If `lspci` exists and nvidia device also exists then we continue.

**Pull request merged**